File: server/fuelboss/models/Units.py

class Units:

    def __init__(self, conf):
        parts = conf.split(',')
        self.abbreviation = parts[0]
        self.plural = parts[1]
        self.fromLiters = float(parts[2])
        self.precision = int(parts[3])
        
    def toUnits(self, liters):
        return liters * self.fromLiters
        
    def format(self, liters):
        return ('{:0.' + self.precision + 'f}').format(self.toUnits(liters))
        
    def toDict(self):
        return {
            'abbreviation': self.abbreviation,
            'plural': self.plural,
            'fromLiters': self.fromLiters,
            'precision': self.precision,
        }
        

File: client/src/units.js

export default {

    toUnits(liters, conf) {
        return liters * conf.fromLiters
    },

    format(liters, conf, appendUnits = true) {
        let str = (liters * conf.fromLiters).toFixed(conf.precision)
        if (appendUnits)
            str += ' ' + conf.abbreviation
        return str
    },
    
}

